atlasdb-cassandra/src/test/java/com/palantir/atlasdb/cassandra/backup/TransactionAborterTest.java

```diff
@@ -0,0 +1,307 @@
+public class TransactionAborterTest {
+    private static final long BACKUP_TIMESTAMP = 123;
+
+    private static final FullyBoundedTimestampRange TIMESTAMP_RANGE =
+            FullyBoundedTimestampRange.of(Range.closed(AtlasDbConstants.STARTING_TS, 200L));
+
+    private static final String TXN_TABLE_NAME = "txn_table";
+    private static final String KEYSPACE = "keyspace";
+
+    @Mock
+    private CqlSession cqlSession;
+
+    @Mock
+    private CassandraKeyValueServiceConfig config;
+
+    @Mock
+    private TransactionsTableInteraction transactionInteraction;
+
+    @Mock
+    private PreparedStatement preparedAbortStatement;
+
+    @Mock
+    private Statement abortStatement;
+
+    @Mock
+    private PreparedStatement preparedCheckStatement;
+
+    @Mock
+    private Statement checkStatement;
+
+    @Mock
+    private TableMetadata tableMetadata;
+
+    private TransactionAborter transactionAborter;
+
+    @Before
+    public void before() {
+        when(transactionInteraction.getTransactionsTableName()).thenReturn(TXN_TABLE_NAME);
+
+        when(abortStatement.getSerialConsistencyLevel()).thenReturn(ConsistencyLevel.SERIAL);
+        when(transactionInteraction.bindAbortStatement(eq(preparedAbortStatement), any()))
+                .thenReturn(abortStatement);
+
+        when(checkStatement.getSerialConsistencyLevel()).thenReturn(ConsistencyLevel.SERIAL);
+        when(transactionInteraction.bindCheckStatement(eq(preparedCheckStatement), any()))
+                .thenReturn(checkStatement);
+
+        when(config.getKeyspaceOrThrow()).thenReturn(KEYSPACE);
+
+        when(tableMetadata.getName()).thenReturn(TXN_TABLE_NAME);
+        KeyspaceMetadata keyspaceMetadata = mock(KeyspaceMetadata.class);
+        when(keyspaceMetadata.getTables()).thenReturn(ImmutableList.of(tableMetadata));
+        CqlMetadata cqlMetadata = mock(CqlMetadata.class);
+        when(cqlMetadata.getKeyspaceMetadata(KEYSPACE)).thenReturn(keyspaceMetadata);
+        when(cqlSession.getMetadata()).thenReturn(cqlMetadata);
+
+        Statement mockStatement = mock(Statement.class);
+        doReturn(ImmutableList.of(mockStatement))
+                .when(transactionInteraction)
+                .createSelectStatementsForScanningFullTimestampRange(any());
+        ResultSet selectResponse = createSelectResponse(ImmutableList.of());
+        when(cqlSession.execute(mockStatement)).thenReturn(selectResponse);
+
+        doReturn(mock(PreparedStatement.class)).when(transactionInteraction).prepareAbortStatement(any(), any());
+        doReturn(mock(PreparedStatement.class)).when(transactionInteraction).prepareCheckStatement(any(), any());
+
+        transactionAborter = new TransactionAborter(cqlSession, config);
+    }
+
+    @Test
+    public void willNotRunAbortWhenNothingToAbort() {
+        transactionAborter.abortTransactions(BACKUP_TIMESTAMP, List.of(transactionInteraction));
+
+        verify(cqlSession, times(1)).execute(any(Statement.class));
```

Comment:
This is a bit obscure -- we have to know that internally one execute means we didn't abort anything because we still execute a select. Maybe verify that the statement run is the select statement and no other interactions to make it obvious?